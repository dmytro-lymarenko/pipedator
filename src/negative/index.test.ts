import { negative } from './index';

describe('negative()', () => {
	it('should validate', () => {
		expect(negative().validate(-4)).toBeNull();
	});

	it('should fail', () => {
		expect(negative().validate(0)).toMatchSnapshot();
		expect(negative().validate(4)).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(negative('Custom message').validate(1)).toMatchSnapshot();
	});
});
