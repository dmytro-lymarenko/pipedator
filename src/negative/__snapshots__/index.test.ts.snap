// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`negative() should fail 1`] = `
Object {
  "children": null,
  "message": "Value should be negative",
  "path": "",
}
`;

exports[`negative() should fail 2`] = `
Object {
  "children": null,
  "message": "Value should be negative",
  "path": "",
}
`;

exports[`negative() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
